gax/src/main/java/com/google/api/gax/retrying/RetryingContext.java

```diff
@@ -43,4 +46,18 @@
   /** Returns the {@link ApiTracer} associated with the current operation. */
   @Nonnull
   ApiTracer getTracer();
+
+  /**
+   * Returns the {@link RetrySettings} to use with this context, or <code>null</code> if the default
+   * {@link RetrySettings} should be used.
```

Comment:
could it just return the default settings instead of null?

Reply:
It does not have the default settings (default settings in this case means the default configured settings for a specific RPC), and the zero value of `RetrySettings` has a meaning (it indicates that the retries should never stop). Also, the common practice in gax is to use `null` to indicate that the default should not be overridden.